build-chef-tarball

```diff
@@ -29,10 +31,11 @@ ruby_url='http://cache.ruby-lang.org/pub/ruby/2.2/ruby-2.2.1.tar.gz'
 chef_version='12.2.1'
 chef_dir='/opt/chef'
 jobs=4
-while getopts 'c:d:hj:r:v:' option; do
+while getopts 'c:d:ghj:r:v:' option; do
 	case "$option" in
 		c) chef_dir=$OPTARG;;
 		d) cache_dir=$OPTARG;;
+		g) GZ=true; WITHOUT_GMP=--without-gmp;;
```

Comment:
as a final note, it's also possible that `WITHOUT_GMP` could have been defined by the incoming environment, so it may be cleaner to use an approach like this (code sliced up for brevity):

``` bash
gz=false
ruby_configure_opts=()
while getopts ... "$option"; do
  case "$option" in
    g) gz=true; ruby_configure_opts+=(--without-gmp);;
  esac
done

...

if $gz; then
  foo
fi

...

./configure "${ruby_configure_opts[@]}" \
  --with-foo=bar \
  --without-baz \
  etc ...
```

- `$gz` need not be quoted as its value is controlled 100% by this program
- more options in the future could add options to ruby `./configure` by appending to the array, the array will expand to nothing if it is empty